cordformation/src/main/kotlin/net/corda/plugins/Cordformation.kt

```diff
@@ -37,11 +37,16 @@ class Cordformation : Plugin<Project> {
          * @param jarName The name of the JAR you wish to access.
          * @return A file handle to the file in the JAR.
          */
+        private const val VERSION_ID_PATTERN = "\\d\\.\\d(-SNAPSHOT|-RC\\d{2}|.\\d{8})?"
+        private val CORDA_JAR_REGEX = ".*corda-$VERSION_ID_PATTERN(-.*)?\\.jar\$".toRegex(RegexOption.IGNORE_CASE)
+        private val CORDA_ENT_JAR_REGEX = ".*corda-enterprise-$VERSION_ID_PATTERN(-.*)?\\.jar\$".toRegex(RegexOption.IGNORE_CASE)
+
         fun verifyAndGetRuntimeJar(project: Project, jarName: String): File {
```

Comment:
I just meant defining the regexp pattern outside the _filter_, not outside the function :wink:. The original approach of
```kotlin
$jarName(-enterprise)?-$releaseVersion\\.jar
```
was fine. We only needed to extend it to allow for an arbitrary classifier.
```kotlin
    @Test
    fun regexp() {
        val jarName = "corda"
        val releaseVersion = "1.2.3.4"
        val pattern = "\\Q$jarName\\E(-enterprise)?-\\Q$releaseVersion\\E(-.+)?\\.jar\$".toRegex()
        println(pattern.toString())

        assertTrue(pattern.matches("corda-1.2.3.4.jar"))
        assertTrue(pattern.matches("corda-enterprise-1.2.3.4.jar"))
        assertTrue(pattern.matches("corda-1.2.3.4-pants.jar"))
        assertTrue(pattern.matches("corda-enterprise-1.2.3.4-wibble.jar"))
    }
```

Reply:
And keep the `IGNORE_CASE` option too, seeing as both MacOS and Windows have case-insensitive filesystems:
```kotlin
    @Test
    fun regexp() {
        val jarName = "corda"
        val releaseVersion = "1.2.3.4"
        val pattern = "\\Q$jarName\\E(-enterprise)?-\\Q$releaseVersion\\E(-.+)?\\.jar\$".toRegex(RegexOption.IGNORE_CASE)

        assertTrue("/absolute/path/corda-1.2.3.4.jar".contains(pattern))
        assertTrue("other/path/corda-enterprise-1.2.3.4.jar".contains(pattern))
        assertTrue("my/path/corda-1.2.3.4-pants.jar".contains(pattern))
        assertTrue("big/path/corda-enterprise-1.2.3.4-wibble.jar".contains(pattern))
        assertFalse("corda-1.2.4-pants.jar".contains(pattern))
        assertFalse("corda-enterprise-1.2.4-wibble.jar".contains(pattern))
    }
```

Reply:
Did you want to keep the `RegexOption.IGNORE_CASE`?